### DISCUSSION TITLE
Background image with encoded svg

### DISCUSSION THREAD

**Discussion by noahw3** [tailwindlabs/tailwindcss#12772] [Help] [answered]
I'm attempting to set a background image using an arbitrary value that is an svg. I'm not entirely sure where I've gone wrong, but I can't seem to get it to work. I've tried setting it inline and setting it as part of the theme, and neither work. No class generates at all.

Here's what I'm using: `bg-[url("data:image/svg+xml;charset=UTF-8,<svg width='1px' height='1px' xmlns='http://www.w3.org/2000/svg'><rect width='100%' height='100%' style='fill:rgba(0, 54, 255, 0.13);'/></svg>")]`

https://github.com/tailwindlabs/tailwindcss/discussions/12322 Seems like a very similar problem, if not the same. I also found [this plugin](https://github.com/AndersNielsen85/tailwindcss-bg-svg/tree/master), which looks like it only works because it's writing the classes itself.

**Comment by wongjn** [comment]
You'd need to URL-encode the SVG:
```lang-none
%3Csvg width='1px' height='1px' xmlns='http://www.w3.org/2000/svg'%3E%3Crect width='100%25' height='100%25' style='fill:rgba(0, 54, 255, 0.13);'/%3E%3C/svg%3E
```

[See it working in this Tailwind Play](https://play.tailwindcss.com/aXDBNAxaqZ).

**Reply by noahw3** [reply, marked as answer]
Ah thank you for pointing me in the right direction. It actually does seem to work without url encoding as well, turns out I just needed to recompile as changes to my tailwind config weren't being picked up dynamically.

[Updated Tailwind Play](https://play.tailwindcss.com/pOI6UrKkas?file=config)

It does only work by extending the tailwind theme though, trying to use an inline arbitrary value does not (probably something to do with trying to create an invalid class name).

**Comment by ishaiavrahami** [comment]
Is this only possible by extending?? Im also trying to do this inline in my class
